### ISSUE TITLE
adb pair should be a String

### ISSUE THREAD

**Issue by samuel-cavalcanti** [cocool97/adb_client#29]
Hi, I have found a bug where the pair code starts with 0, as shown below:

<img src= "https://github.com/user-attachments/assets/c9d1f309-68c7-4af9-b007-1535419e812a" height="500"/>

if you parse the string "091102" as a u32 `let code = ""091102".parse().unwrap();` the number will be incorrect

Hi, looking in adb source code: [pairing_connection_test][pairing_connection_test.cpp], [pairing_server.cpp][pairing_server.cpp], [pairing_connection.cpp][pairing_connection.cpp]
 I found that the password or the pair code should be a vector of 8 bytes (Vec<u8>)

```c++
// pairing_connection_test.cpp at line 161
TEST_F(AdbWifiPairingConnectionTest, ClientCreation) {
    // All parameters bad
    auto client = PairingClient::create({}, client_info_, {}, {}, "");
    EXPECT_EQ(nullptr, client);
    // Bad password
    client = PairingClient::create({}, client_info_, {0x01}, {0x01}, "127.0.0.1");
     ...
    // Valid params
    client = PairingClient::create({0x01}, client_info_, {0x01}, {0x01}, "127.0.0.1");
    EXPECT_NE(nullptr, client);
}
TEST_F(AdbWifiPairingConnectionTest, SmokeValidPairing) {
    std::vector<uint8_t> pswd{0x01, 0x03, 0x05, 0x07};
    initPairing(pswd, pswd);
   .....
```

```c++  
// pairing_server.cpp   at line 397
 std::unique_ptr<PairingServer> PairingServer::create(const Data& pswd, const PeerInfo& peer_info,
                                                     const Data& cert, const Data& priv_key,
                                                     int port) {
    if (pswd.empty() || cert.empty() || priv_key.empty() || port <= 0) {
        return nullptr;
 adb_pairing_auth_test   }
....
```

```c++
// pairing_connection.cpp at line 57
struct PairingConnectionCtx {
  public:
    using Data = std::vector<uint8_t>; // <-- Data is defined as a vector of u8
    using ResultCallback = pairing_result_cb;
    enum class Role {
        Client,
        Server,
    };

```
[pairing_connection_test.cpp]: https://android.googlesource.com/platform/packages/modules/adb/+/refs/heads/main/client/pairing/tests/pairing_connection_test.cpp#161

[pairing_connection.cpp]: https://android.googlesource.com/platform/packages/modules/adb/+/refs/heads/main/pairing_connection/pairing_connection.cpp#57

[pairing_server.cpp]: https://android.googlesource.com/platform/packages/modules/adb/+/refs/heads/main/client/pairing/tests/pairing_server.cpp#396

**Comment by samuel-cavalcanti** [comment]
I tried to implement as Vec<u8>, but it's challenging with many edge cases. A Rust String works nicely instead :)

**Comment by cocool97** [comment]
Thanks for pointing this issue ! Can you add your fix in a PR ;) ?

**Comment by cocool97** [comment]
Closed and merged in 1.0.6
